### DISCUSSION TITLE
How To Get Raylib working in android Studio

### DISCUSSION THREAD

**Discussion by 1NEGROUP** [raysan5/raylib#2864] [Q&A]
So I Have been trying to figure out how to do this, but I am Having problems with the CMakeFile in AndroidStudio and Kinda of confused on how to set things up. Here is the [CMakeFile.txt](https://paste.myst.rs/hjr62dxq)

Here is the Error i get in Android Studio 
_CMake Error at CMakeLists.txt:16 (find_package):
By not providing "Findraylib.cmake" in CMAKE_MODULE_PATH this project has
asked CMake to find a package configuration file provided by "raylib", but
CMake did not find one._

**Comment by gen2brain** [comment]
You are using `find_package`, but raylib doesn't provide `Findraylib.cmake`, or raylib-config.cmake, see https://github.com/raysan5/raylib/issues/471. It does provide `.pc` file, so you can use pkg-config in CMake instead, something like this, I am not really an expert in CMake:
```
find_package(PkgConfig REQUIRED)
pkg_check_modules(raylib REQUIRED raylib)
```
Of course, you must have raylib compiled/installed for Android, and you should tell pkg-config where to find them, or use a standalone Android toolchain where you would install the library, etc.

What advantages do you think AndroidStudio could provide you? The raylib is using NativeActivity and "native_app_glue", IMO you are better off with some good C editor/IDE instead. There is a Makefile that can build apk for you, and there was some contribution that works with bash and tools only, not even `make` is needed.

**Reply by 1NEGROUP** [reply]
> You are using `find_package`, but raylib doesn't provide `Findraylib.cmake`, or raylib-config.cmake,

What do you mean its not provided, There is a raylib-config.cmake file in the cmake folder?
I will try the new Cmake instructions and let you know if it works

**Reply by 1NEGROUP** [reply]
> There is a Makefile that can build apk for you, and there was some contribution that works with bash and tools only, not even `make` is needed.

Thank You Thats Kinda what i need. Although the benefit would be dealing with the resolution a lot easier in AndroidStudio
I would actually Just prefer a text editor if i could, and I guess i will just have to find an android emulator.

So Can i just start coding in C++ then just build the apk, or do i need to set somethng else up first?

**Reply by gen2brain** [reply]
Check https://github.com/raysan5/raylib/wiki/Working-for-Android and prepare your environment. As for the emulator, I used genymotion before, based on vbox, and is accelerated, but qemu will do just fine. I just use an old android device I have when I am testing apk, no need for the emulator.
